linkerd/app/outbound/src/http/concrete.rs

```diff
@@ -77,31 +109,241 @@ impl<N> Outbound<N> {
+        }
+    }
+}
+
+impl<T> svc::Param<metrics::EndpointLabels> for Endpoint<T>
+where
+    T: svc::Param<Option<profiles::LogicalAddr>>,
+{
+    fn param(&self) -> metrics::EndpointLabels {
+        metrics::EndpointLabels::from(svc::Param::<metrics::OutboundEndpointLabels>::param(self))
+    }
+}
+
+impl<T> svc::Param<tls::ConditionalClientTls> for Endpoint<T> {
+    fn param(&self) -> tls::ConditionalClientTls {
+        if self.is_local {
+            return tls::ConditionalClientTls::None(tls::NoClientTls::Loopback);
+        }
+
+        // If we're transporting an opaque protocol OR we're communicating with
+        // a gateway, then set an ALPN value indicating support for a transport
+        // header.
+        let use_transport_header = self.metadata.opaque_transport_port().is_some()
+            || self.metadata.authority_override().is_some();
+        self.metadata
+            .identity()
+            .cloned()
+            .map(move |server_id| {
+                tls::ConditionalClientTls::Some(tls::ClientTls {
+                    server_id,
+                    alpn: if use_transport_header {
+                        use linkerd_app_core::transport_header::PROTOCOL;
+                        Some(tls::client::AlpnProtocols(vec![PROTOCOL.into()]))
+                    } else {
+                        None
+                    },
+                })
+            })
+            .unwrap_or(tls::ConditionalClientTls::None(
+                tls::NoClientTls::NotProvidedByServiceDiscovery,
+            ))
+    }
+}
+
+impl<T> svc::Param<normalize_uri::DefaultAuthority> for Endpoint<T>
+where
+    T: svc::Param<Option<profiles::LogicalAddr>>,
+{
+    fn param(&self) -> normalize_uri::DefaultAuthority {
+        if let Some(profiles::LogicalAddr(ref a)) = self.parent.param() {
+            return normalize_uri::DefaultAuthority(Some(
+                a.to_string()
+                    .parse()
+                    .expect("Address must be a valid authority"),
+            ));
+        }
+
+        normalize_uri::DefaultAuthority(Some(
+            self.addr
+                .to_string()
+                .parse()
+                .expect("Address must be a valid authority"),
+        ))
+    }
+}
+
+impl<T> svc::Param<http::Version> for Endpoint<T>
+where
+    T: svc::Param<http::Version>,
+{
+    fn param(&self) -> http::Version {
+        self.parent.param()
+    }
+}
+
+impl<T> svc::Param<client::Settings> for Endpoint<T>
+where
+    T: svc::Param<http::Version>,
+{
+    fn param(&self) -> client::Settings {
+        match self.param() {
+            http::Version::H2 => client::Settings::H2,
+            http::Version::Http1 => match self.metadata.protocol_hint() {
+                ProtocolHint::Unknown => client::Settings::Http1,
+                ProtocolHint::Http2 => client::Settings::OrigProtoUpgrade,
+            },
+        }
+    }
+}
+
+// TODO(ver) move this into the endpoint stack?
```

Comment:
is this comment still relevant here? looks like it was pasted from the earlier code?